### PR TITLE
Roya internship

### PR DESCRIPTION
**Pull request by Royamosavipour**
Hei I made new changes, please guide me and tell me your opinion

### CODE REVIEW

File: app/src/components/Project/Content/AllCardsContent.tsx [hunk 1]
```diff
@@ -101,7 +101,7 @@ export default function AllCardsContent({
                 <RiArrowLeftSLine onClick={handleScrollForward} />
               </div>
             </div>
-            <div ref={scrollContainerRef} className='flex overflow-x-scroll'>
+            <div ref={scrollContainerRef} className='flex overflow-x-scroll snap-x snap-mandatory'>
               {filteredCards
                 .sort(({ createdAt: a }, { createdAt: b }) => b - a)
                 .map((card, index) => (
```
Comment (eugeek) [inline app/src/components/Project/Content/AllCardsContent.tsx:107]: please add the `className='snap-center'` to each rendered item. It should work =)

File: app/src/components/Chats/ChatPreview.tsx [hunk 2]
```diff
@@ -61,77 +66,102 @@ export default function ChatPreview({
   const firstMessage = sortedMessages ? sortedMessages[0] : null
   const lastMessage = sortedMessages && sortedMessages.length > 1 ? sortedMessages.at(-1) : null
   const repliesCount = sortedMessages ? sortedMessages.length - 2 : 0
+  const showName =
```
Comment (eugeek) [inline app/src/components/Chats/ChatPreview.tsx:69]: ?

File: app/src/components/Chats/ChatPreview.tsx [hunk 3]
```diff
@@ -61,77 +66,102 @@ export default function ChatPreview({
...
+    // const messageUser = lastMessage ? chatUsers[lastMessage.authorId] : null
```
Comment (eugeek) [inline app/src/components/Chats/ChatPreview.tsx:70]: remove it too

File: app/src/components/Chats/ChatPreview.tsx [hunk 4]
```diff
@@ -61,77 +66,102 @@ export default function ChatPreview({
...
+      <div
+        className={`w-2 h-full border-r-2 border-border-color ${
+          disableChat ? 'bg-[#EFEFEF]' : 'bg-[#DBF7CA]'
+        } `}
+      ></div>
+      <div
+        className={`w-full flex border-border-color hover:cursor-pointer hover:bg-gray-100 pr-7  py-3 ${
+          disableChat ? 'pointer-events-none opacity-50' : ''
+        }`}
+        onClick={disableChat ? undefined : onClick}
```
Comment (eugeek) [inline app/src/components/Chats/ChatPreview.tsx:94]: if you have `undefined` the better way will be `isChatDisabled && onClick`

File: app/src/components/Card/Sections/Discussions.tsx [hunk 5]
```diff
@@ -24,6 +24,8 @@ export default function Discussions({ users }: { users: IUser[] }) {
       chat.content.toLowerCase().includes(search.toLowerCase()),
     )
     setFilteredChats(updatedChats)
+
+    console.log(filteredChats);
```
[the remark this answers is not in the document]
Comment (Royamosavipour) [reply app/src/components/Card/Sections/Discussions.tsx:28]: Yes, of course, and thank you for this point

File: app/src/components/Chats/ChatPreview.tsx [hunk 6]
```diff
@@ -61,77 +66,102 @@ export default function ChatPreview({
...
-              <i
-                onClick={(e) => e.stopPropagation()}
-                className='z-40 ri-more-2-fill text-2xl absolute top-1 right-5'
+    // <div
+    //   className='w-full flex border-border-color hover:cursor-pointer hover:bg-gray-100 pr-7  py-3'
+    //   onClick={onClick}
+    // >
+    <div className='flex w-full'>
+      <div
+        className={`w-2 h-full border-r-2 border-border-color ${
```
Comment (eugeek) [inline app/src/components/Chats/ChatPreview.tsx:86]: let's use `clsx` for it
should be like that:
`className={clsx('w-2 h-full border-r-2 border-border-color', isChatDisabled ? 'bg-[#EFEFEF]' : 'bg-[#DBF7CA]')}`

File: app/src/components/Project/Content/AllCardsContent.tsx (hunk 1 above)
Comment (Royamosavipour) [reply app/src/components/Project/Content/AllCardsContent.tsx:107]: sorry egueek i should remove 'snap-x snap-mandatory'

File: app/src/components/Chats/ChatPreview.tsx (hunk 2 above)
Comment (Royamosavipour) [reply app/src/components/Chats/ChatPreview.tsx:69]: I made mistake now i removed it

File: app/src/components/Chats/ChatPreview.tsx [hunk 7]
```diff
@@ -61,77 +66,102 @@ export default function ChatPreview({
...
+          disableChat ? 'bg-[#EFEFEF]' : 'bg-[#DBF7CA]'
+        } `}
+      ></div>
+      <div
+        className={`w-full flex border-border-color hover:cursor-pointer hover:bg-gray-100 pr-7  py-3 ${
```
Comment (eugeek) [inline app/src/components/Chats/ChatPreview.tsx:91]: the same `clsx`

File: app/src/components/Chats/ChatPreview.tsx [hunk 8]
```diff
@@ -61,77 +66,102 @@ export default function ChatPreview({
...
-  // const messageUser = lastMessage ? chatUsers[lastMessage.authorId] : null
+    useEffect(() => {
+      setUnreadMessages(getUnreadMessagesCount(chat.id))
+      // console.log(chatUsers[firstMessage.authorId].name)
+    }, [unreadChats])
 
-  useEffect(() => {
-    setUnreadMessages(getUnreadMessagesCount(chat.id))
-  }, [unreadChats])
+  const disableChat = unreadMessages === 0
```
[the remark this answers is not in the document]
Comment (Royamosavipour) [reply app/src/components/Chats/ChatPreview.tsx:77]: yes sure i change it with isChatDisabled

File: app/src/components/Chats/ChatPreview.tsx (hunk 7 above)
Comment (Royamosavipour) [reply app/src/components/Chats/ChatPreview.tsx:91]: Thank you for this way

File: app/src/components/Project/Content/AllCardsContent.tsx (hunk 1 above)
Comment (eugeek) [reply app/src/components/Project/Content/AllCardsContent.tsx:107]: no

File: app/src/components/Chats/ChatPreview.tsx [hunk 9]
```diff
@@ -66,32 +67,27 @@ export default function ChatPreview({
...
       <div
-        className={`w-full flex border-border-color hover:cursor-pointer hover:bg-gray-100 pr-7  py-3 ${
-          disableChat ? 'pointer-events-none opacity-50' : ''
-        }`}
-        onClick={disableChat ? undefined : onClick}
+        className={clsx(
+          'w-full flex border-border-color hover:cursor-pointer hover:bg-gray-100 pr-7 py-3',
+          isChatDisabled ? 'pointer-events-none opacity-50' : '',
+        )}
+        onClick={isChatDisabled && onClick}
```
Comment (eugeek) [inline app/src/components/Chats/ChatPreview.tsx:90]: !isChatDisabled* my bad 🫠

File: app/src/components/Project/Content/AllCardsContent.tsx [hunk 10]
```diff
@@ -101,7 +101,7 @@ export default function AllCardsContent({
                 <RiArrowLeftSLine onClick={handleScrollForward} />
               </div>
             </div>
-            <div ref={scrollContainerRef} className='flex overflow-x-scroll snap-x snap-mandatory'>
+            <div ref={scrollContainerRef} className='flex overflow-x-scroll snap-x snap-mandatory snap-center'>
```
Comment (eugeek) [inline app/src/components/Project/Content/AllCardsContent.tsx:104]: not here - to children. Remain the snap-x and snap-mandatory at parent + add snap-center to children
https://tailwindcss.com/docs/scroll-snap-type

Comment (Royamosavipour) [reply app/src/components/Project/Content/AllCardsContent.tsx:104]: sorry eugeek I updated it but didn't resolve the problem

File: app/src/components/Chats/ChatPreview.tsx (hunk 9 above)
Comment (Royamosavipour) [reply app/src/components/Chats/ChatPreview.tsx:90]: sorry eugeek I did not understand what to do in this part

Comment (eugeek) [reply app/src/components/Chats/ChatPreview.tsx:90]: isChatDisabled - when chat is disabled and onClick should not work.
So !isChatDisabled - chat is not disabled and onClick should work

Comment (eugeek) [reply app/src/components/Chats/ChatPreview.tsx:90]: if chat is disabled how onclick will work?

**Pull request merged**